pulsar-functions/localrun/src/main/java/org/apache/pulsar/functions/LocalRunner.java

```diff
@@ -418,11 +428,24 @@ private void startThreadedMode(org.apache.pulsar.functions.proto.Function.Functi
                                            int parallelism, int instanceIdOffset, String serviceUrl,
                                            String stateStorageServiceUrl, AuthenticationConfig authConfig,
                                            String userCodeFile) throws Exception {
+        SecretsProvider secretsProvider;
+        if (secretsProviderClassName != null) {
+            if (secretsProviderClassName.equals(ClearTextSecretsProvider.class.getName())) {
```

Comment:
Just create an instance using the class name. Use method:

https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/util/Reflections.java#L64

Reply:
Done